compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -5114,6 +5117,97 @@ public void visit(BLangForeach foreach) {
         }
     }
 
+    private boolean canEliminateIterator(BLangForeach loop) {
+        BLangExpression collection = loop.collection;
+        if (collection.getKind() == NodeKind.BINARY_EXPR) {
+            // range expression
+            return true;
+        }
+        TypeKind kind = collection.getBType().getKind();
+        return kind == TypeKind.ARRAY || kind == TypeKind.TUPLE;
+    }
+
+    private BLangBlockStmt desugarForeachToWhileWithoutIterator(BLangForeach foreach) {
```

Comment:
Is it possible to add a comment showing the before desugar and after desugar code similar to`desugarForeachToWhileWithIterator()` method?

https://github.com/ballerina-platform/ballerina-lang/blob/ceb5155986a87aa68cea016262534f9fe9aa097d/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java#L5315-L5340

Reply:
Fixed